src/lib/components/ActionButton.tsx

```diff
@@ -71,10 +71,14 @@ export default function ActionButton({
   onUpdate,
   children,
 }: ActionButtonProps) {
+  const textColor = useMemo(
```

Comment:
Sorry for the churn but I don't know that `useMemo` adds any value here. Can't you just compute the textColor directly?

Reply:
Same with other `useMemo` uses.